src/Compilers/VisualBasic/Portable/Declarations/DeclarationTreeBuilder.vb

```diff
@@ -549,12 +547,28 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
             Return children.ToImmutableAndFree()
         End Function
 
-        Private Function GetNonTypeMemberNames(members As SyntaxList(Of StatementSyntax), ByRef declFlags As SingleTypeDeclaration.TypeDeclarationFlags) As String()
+        ''' <summary>
+        ''' Pool of builders used to create our member name sets.  Importantly, these use 
+        ''' <see cref="CaseInsensitiveComparison.Comparer"/> so that name lookup happens in an
+        ''' appropriate manner for VB identifiers. This allows fast member name O(log(n)) even if
+        ''' the casing doesn't match.
+        ''' </summary>
```

Comment:
note: we have lot of case sensitivty tests for VB (including directly in symbol search) to validate this behavior.